### ISSUE TITLE
System.Windows.Forms.Design.ControlDesigner does not handle mouse messages on adorner window such as resizing 

### ISSUE THREAD

**Issue by dmitry-medvedev** [dotnet/winforms#9160]
### .NET version

.NET Core 7

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Works fine for .NET 6

### Issue description

We use .NET System.Windows.Forms.Design services for the run-time implementation of the Form designer functionality.
Form designer works fine in .Net core 6 (and .net 4.62), allowing placing and resizing controls on the design surface. However, resizing does not work with .net 7.

### Steps to reproduce

It would be hard to give a reproducible scenario, but by tracing the source code, we've found the following:

.NET 6
BehaviorService.cs
...
private bool ProcessMouseMessage(IntPtr hWnd, Interop.User32.WM msg, int x, int y)
    Point point = new Point(x, y);
    Interop.User32.MapWindowPoints(IntPtr.Zero, adornerWindow.Handle, ref point, 1U);
    Message message = Message.Create(hWnd, msg, (IntPtr)0, Interop.PARAM.FromLowHigh(p.Y, p.X));

.NET 7

BehaviorService.cs
...
private bool ProcessMouseMessage(IntPtr hWnd, Interop.User32.WM msg, int x, int y)
    Point point = new Point(x, y);
    adornerWindow.PointToClient(p); // should be p =adornerWindow.PointToClient(p);
    Message message = Message.Create(hWnd, msg, (IntPtr)0, Interop.PARAM.FromLowHigh(p.Y, p.X));

adornerWindow.PointToClient internally uses MapWindowPoints, but its result is ignored. This is causing sending screen coordinates to the adornerWindow and clearing _hitTestedGlyph field, which results in adorners not behaving as they should.
The same applies to the designer action glyph (displayed at the top right corner of the selected control) - it does nothing on our tests.

**Comment by dmitrii-drobotov** [comment]
We fixed it in https://github.com/dotnet/winforms/pull/8522 for .NET 8 but didn't backport to .NET 7. Given that the fix for .NET 7 is pretty simple (it needs only [MouseHook part](https://github.com/dotnet/winforms/pull/8522/files#diff-fc2b19366f583f6bfd460c97f226cb4bfe354714d89335c6cf2dffc4db8cbb2d)), we might consider backporting it now.

CC @dreddy-work @merriemcgaw

**Comment by dmitry-medvedev** [comment]
Great, thank you!

**Comment by dmitrii-drobotov** [comment]
The fix will be available in .NET 7.0.8.

**Comment by dmitry-medvedev** [comment]
Hi

It looks like the fix has not been included in the 7.0.8 yet:


Point p = new Point(x, y);  
adornerWindow.PointToClient(p);

here's a full decompiled code of the ProcessMouseMessage:

// C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\7.0.8\System.Windows.Forms.Design.dll

```
public sealed class BehaviorService : IDisposable
...
	private bool ProcessMouseMessage(IntPtr hWnd, Interop.User32.WM msg, int x, int y)
				{
					if (this._processingMessage)
					{
						return false;
					}
					foreach (BehaviorService.AdornerWindow adornerWindow in BehaviorService.AdornerWindow.s_adornerWindowList)
					{
						if (adornerWindow.DesignerFrameValid)
						{
							this._currentAdornerWindow = adornerWindow;
							IntPtr handle = adornerWindow.DesignerFrame.Handle;
							if (adornerWindow.ProcessingDrag || (hWnd != handle && Interop.User32.IsChild(new HandleRef(this, handle), hWnd).IsTrue()))
							{
								uint num;
								Interop.User32.GetWindowThreadProcessId(hWnd, out num);
								if (num != this._thisProcessID)
								{
									return false;
								}
								try
								{
									this._processingMessage = true;
									Point p = new Point(x, y);  
									adornerWindow.PointToClient(p);
									Message message = Message.Create(hWnd, msg, (IntPtr)0, Interop.PARAM.FromLowHigh(p.Y, p.X));
									if (message.Msg == 513)
									{
										this._lastLButtonDownTimeStamp = Interop.User32.GetMessageTime();
									}
									else if (message.Msg == 515 && Interop.User32.GetMessageTime() == this._lastLButtonDownTimeStamp)
									{
										return true;
									}
									if (!adornerWindow.WndProcProxy(ref message, p.X, p.Y))
									{
										return true;
									}
									break;
								}
								finally
								{
									this._processingMessage = false;
								}
							}
						}
					}
					return false;
				}
```

Kind regards,
Dmitry

**Comment by dmitry-medvedev** [comment]
Thank you for confirming.
